StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/PayWithLinkController-New.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  PayWithNativeLinkController.swift
+//  StripePaymentSheet
+//
+//  Created by Ramon Torres on 7/18/22.
+//  Copyright © 2022 Stripe, Inc. All rights reserved.
+//
+
+@_spi(STP) import StripeCore
+@_spi(STP) import StripePayments
+@_spi(STP) import StripeUICore
+import UIKit
+
+/// Standalone Link controller
+@available(iOSApplicationExtension, unavailable)
+@available(macCatalystApplicationExtension, unavailable)
+final class PayWithNativeLinkController {
+
+    typealias CompletionBlock = PaymentSheetResultCompletionBlock
+
+    private let paymentHandler: STPPaymentHandler
+
+    private var completion: PaymentSheetResultCompletionBlock?
+
+    private var selfRetainer: PayWithNativeLinkController?
+
+    let intent: Intent
+    let elementsSession: STPElementsSession
+    let configuration: PaymentSheet.Configuration
+
+    init(intent: Intent, elementsSession: STPElementsSession, configuration: PaymentSheet.Configuration) {
+        self.intent = intent
+        self.configuration = configuration
+        self.elementsSession = elementsSession
+        self.paymentHandler = .init(apiClient: configuration.apiClient)
+    }
+
+    func present(completion: @escaping CompletionBlock) {
+        guard
+            let keyWindow = UIApplication.shared.stp_hackilyFumbleAroundUntilYouFindAKeyWindow(),
+            let presentedViewController = keyWindow.findTopMostPresentedViewController()
+        else {
+            assertionFailure("No key window with view controller found")
```

Comment:
1. stpAssertionFailure
2.  in PayWithLinkWebController we use:
```
    var bestWindowForPresentation: UIWindow {
        if let window = presentationVC?.view.window {
            return window
        }
        return UIApplication.shared.stp_hackilyFumbleAroundUntilYouFindAKeyWindow()!
    }
```
maybe we can add a TODO here to pass in a presentationVC rather than rely on `stp_hackilyFumbleAroundUntilYouFindAKeyWindow`

Reply:
After more investigation, I think I'll kill this file for now. It's for a standalone flow that we don't currently offer.